### PR TITLE
try to fix uncaught FailedPayloadsError in consumer

### PR DESCRIPTION
**Pull request by reAsOn2010**
about #386. It‘s a not that elegant fix, but works...

### CODE REVIEW

**Comment by reAsOn2010** [comment]
And this fix not solve the second problem about NotLeaderForPartitionError after the failed broker recover

File: kafka/consumer/simple.py [hunk 1]
```diff
@@ -341,6 +341,8 @@ def _fetch(self):
             for resp in responses:
 
                 try:
+                    if isinstance(resp, Exception):
+                        raise resp
```
Comment (dpkp) [inline kafka/consumer/simple.py]: this check can probably move into the kafka.utils.check_error function

File: kafka/consumer/simple.py [hunk 2]
```diff
@@ -353,6 +355,13 @@ def _fetch(self):
                     # Retry this partition
                     retry_partitions[resp.partition] = partitions[resp.partition]
                     continue
+                except FailedPayloadsError:
```
Comment (dpkp) [inline kafka/consumer/simple.py]: would do `except FailedPayloadsError as e` here

File: kafka/consumer/simple.py [hunk 3]
```diff
@@ -353,6 +355,13 @@ def _fetch(self):
...
+                    log.warning("Failed payloads of %s"
+                                "Resetting partition offset...",
+                                resp.failed_payloads)
+                    # Retry this partition
+                    retry_partitions[resp.failed_payloads.partition] = partitions[resp.failed_payloads.partition]
```
Comment (dpkp) [inline kafka/consumer/simple.py]: .failed_payloads has been changed to .payload .  suggest using `e.payload.partition`

**Comment by dpkp** [comment]
thanks for the PR! I added a few comments, and also would love if you could take a stab at adding a test for this.

**Comment by reAsOn2010** [comment]
... why I mentioned others' commits, or I make a new PR?

**Pull request merged**